girder/molecules/molecules/utilities/molecules.py

```diff
@@ -80,13 +77,7 @@ def create_molecule(data_str, input_format, user, public, gen3d=True,
         if input_format == 'cjson':
             cjson = json.loads(data_str)
 
-        if not cjson and using_2d_format:
-            # Generate 3d coordinates in a background thread
-            if gen3d:
-                schedule_3d_coords_gen(mol_dict, user)
-            # This will be complete other than the cjson
-            return MoleculeModel().create(user, mol_dict, public)
-        else:
+        if cjson or not using_2d_format:
```

Comment:
I think the main purpose of this code block is to convert to cjson if the input is not a 2D format, and it isn't cjson. But we can probably change this in a separate PR...